dandi/metadata.py

```diff
@@ -341,45 +345,128 @@ def extract_sex(metadata):
+    ),
+    (
+        ["jacchus"],
+        None,
+        "http://purl.obolibrary.org/obo/NCBITaxon_9483",
+        "Callithrix jacchus - Common marmoset",
+    ),
+    (
+        ["melanogaster", "fruit fly"],
+        None,
+        "http://purl.obolibrary.org/obo/NCBITaxon_7227",
+        "Drosophila melanogaster - Fruit fly",
+    ),
+]
+
+
+@lru_cache(maxsize=None)
+def parse_purlobourl(url: str, lookup: ty.Tuple[str] = None):
+
+    req = requests.get(url, allow_redirects=True)
+    doc = parseString(req.text)
+    elfound = None
+    for el in doc.getElementsByTagName("Class"):
+        if (
+            "rdf:about" in el.attributes.keys()
+            and el.attributes.getNamedItem("rdf:about").value == url
+        ):
+            elfound = el
+            break
+    if elfound is None:
+        return None
+    values = {}
+    if lookup is None:
+        lookup = ("rdfs:label", "oboInOwl:hasExactSynonym")
+    for key in lookup:
+        elchild = elfound.getElementsByTagName(key)
+        if elchild:
+            elchild = elchild[0]
+            values[key] = elchild.childNodes[0].nodeValue.capitalize()
+    return values
+
+
 def extract_species(metadata):
-    value = metadata.get("species", None)
-    if value is not None and value != "":
-        value = value.lower()
-        if "mouse" in value or value.startswith("mus"):
-            value_id = "http://purl.obolibrary.org/obo/NCBITaxon_10090"
-            value = "House mouse"
-        elif "human" in value or value.startswith("homo"):
-            value_id = "http://purl.obolibrary.org/obo/NCBITaxon_9606"
-            value = "Human"
-        elif "norvegicus" in value:
-            value_id = "http://purl.obolibrary.org/obo/NCBITaxon_10116"
-            value = "Rat; Norway rat; rats; Brown rat"
-        elif "rattus rattus" in value:
-            value_id = "http://purl.obolibrary.org/obo/NCBITaxon_10117"
-            value = "Black rat; Roof rat; House rat"
-        elif "rat" in value:
-            value_id = "http://purl.obolibrary.org/obo/NCBITaxon_10116"
-            value = "Rat; Norway rat; rats; Brown rat"
-        elif "mulatta" in value or "rhesus" in value:
-            value_id = "http://purl.obolibrary.org/obo/NCBITaxon_9544"
-            value = "Rhesus monkey"
-        elif "jacchus" in value:
-            value_id = "http://purl.obolibrary.org/obo/NCBITaxon_9483"
-            value = "Common marmoset"
-        elif "melanogaster" in value or "fruit fly" in value:
-            value_id = "http://purl.obolibrary.org/obo/NCBITaxon_7227"
-            value = "Common fruit fly"
-        elif value.startswith("http"):
-            value_id = value
-            value = None
+    value_orig = metadata.get("species", None)
+    value_id = None
+    if value_orig is not None and value_orig != "":
+        value = value_orig.lower().rstrip("/")
+        if value.startswith("http://purl.obolibrary.org/obo/NCBITaxon_".lower()):
+            for entry in species_map:
```

Comment:
Please do `for names, genus, url, species in species_map` (or some other set of names) instead for better readability.